File: client-app/src/app/models/employee.ts
export interface Employee {
  id: string;
  name: string;
  birthDate: Date | null;
  gender: string;
  email: string;
  cpf: string;
  startDate: Date | null;
  team: string;
}

export class Employee implements Employee {
  constructor(init?: EmployeeFormValues) {
    Object.assign(this, init);
  }
}

export class EmployeeFormValues {
  id?: string = undefined;
  name: string = "";
  birthDate: Date | null = null;
  gender: string = "";
  email: string = "";
  cpf: string = "";
  startDate: Date | null = null;
  team: string = "";

  constructor(employee?: EmployeeFormValues) {
    if (employee) {
      this.id = employee.id;
      this.name = employee.name;
      this.birthDate = employee.birthDate;
      this.gender = employee.gender;
      this.email = employee.email;
      this.cpf = employee.cpf;
      this.startDate = employee.startDate;
      this.team = employee.team;
    }
  }
}

File: Domain/Employee.cs
using System;

namespace Domain
{
    public class Employee
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime BirthDate { get; set; }
        public string Gender { get; set; }
        public string Email { get; set; }
        public string CPF { get; set; }
        public DateTime StartDate { get; set; }
        public string Team { get; set; }

    }
}